qubes-gui-client/src/lib.rs

```diff
@@ -1,7 +1,161 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        assert_eq!(2 + 2, 4);
+/*
+ * The Qubes OS Project, https://www.qubes-os.org
+ *
+ * Copyright (C) 2021  Demi Marie Obenour  <demi@invisiblethingslab.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+//! A client for the Qubes OS GUI protocol.  This client is low-level.
+
+#![forbid(missing_docs)]
+#![forbid(unconditional_recursion)]
+
+use qubes_castable::Castable as _;
+pub use qubes_gui;
+use std::collections::BTreeSet;
+use std::convert::TryInto;
+use std::io;
+use std::num::NonZeroU32;
+use std::task::Poll;
+
+mod buffer;
+
+/// The entry-point to the library.
+#[derive(Debug)]
+pub struct Client {
+    vchan: buffer::RawMessageStream<Option<vchan::Vchan>>,
```

Comment:
With the struct being named `RawMessageStream` now, perhaps a name like `raw` or `inner` would work better?

```suggestion
    raw: buffer::RawMessageStream<Option<vchan::Vchan>>,
```